### PR TITLE
feat: upgrade styled-components to v6

### PR DESCRIPTION
**Pull request by rodleyorosa**
- add `$` prefixes
- update `styled-components` package
- remove `SimpleInterpolation` type with specific return types

Refs: SHELL-107

### CODE REVIEW

File: src/shell/boards/board-container.tsx [hunk 1]
```diff
@@ -444,7 +443,7 @@ export const BoardContainer = (): React.JSX.Element | null => {
 						<Divider style={{ height: '0.125rem' }} />
 						<BoardDetailContainer takeAvailableSpace>
 							{map(boards, (b) => (
-								<AppBoard key={b.id} board={b} />
+								<AppBoard key={b.id} $board={b} />
```
Comment (beawar) [inline src/shell/boards/board-container.tsx]: This should not have $ prefix

File: src/shell/boards/board-tab.tsx [hunk 2]
```diff
@@ -48,9 +47,9 @@ const TabIcon = styled(Icon)`
 	min-width: 1.5rem;
 `;
 
-const CustomText = styled(Text)<{ overflowing: boolean }>`
-	${({ overflowing }): SimpleInterpolation =>
-		overflowing &&
+const CustomText = styled(Text)<{ $overflowing: boolean }>`
+	${({ $overflowing }: { $overflowing: boolean }): ReturnType<typeof css> | false =>
```
Comment (beawar) [inline src/shell/boards/board-tab.tsx]: Type not needed
```suggestion
	${({ $overflowing }): ReturnType<typeof css> | false =>
```

File: src/shell/boards/board.tsx [hunk 3]
```diff
@@ -49,18 +49,18 @@ const BoardViewComponent = ({
 	</AppContextProvider>
 );
 
-export const AppBoard = ({ board }: { board: Board }): React.JSX.Element => {
+export const AppBoard = ({ $board }: { $board: Board }): React.JSX.Element => {
```
Comment (beawar) [inline src/shell/boards/board.tsx]: Do not prefix board prop

**Review by lisaparma — DISMISSED** [review]

**Comment by beawar** [comment]
I think we have finally reached a good point on all involved projects. The PR can be considered stable and reviewable.

**Review by beawar — APPROVED** [review]

**Review by CerDennis — APPROVED** [review]

**Review by CataldoMazzilli — APPROVED** [review]

**Pull request merged**